src/@next/Card/Card.tsx

```diff
@@ -33,16 +36,20 @@ const CardComponent = React.forwardRef<HTMLDivElement, CardProps>(function Card(
   }: CardProps,
   ref
 ) {
-  const headingMarkup = (
-    <Typography as="div" variant="body2">
+  const headingMarkup = PRIMITIVE_TYPES.includes(typeof heading) ? (
+    <Typography as="div" variant="subtitle1">
       {heading}
     </Typography>
+  ) : (
+    <StyledCustomHeader>{heading}</StyledCustomHeader>
   );
 
-  const subHeadingMarkup = (
+  const subHeadingMarkup = PRIMITIVE_TYPES.includes(typeof subheading) ? (
```

Comment:
don't think we need this condition anymore since the Typography is a `div` which can serve as a neutral wrapper

Reply:
The reason I make it conditional is because, when I delete it like this
```tsx
const headingMarkup = (
  <Typography as="div" variant="subtitle1">
    {heading}
  </Typography>
);

const subHeadingMarkup = (
  <Typography as="div" variant="subtitle2">
    {subheading}
  </Typography>
);
```
There are margins on top and bottom of the custom heading/subheading, resulting on this UI:
<img width="994" alt="image" src="https://user-images.githubusercontent.com/52077891/221114101-b9bd2b46-ed78-4771-a4f1-e7dceaceb7bb.png">

I am not sure why this happen

I created a new styled div to mitigate this issue by applying margin: 0,
```ts
export const StyledCustomHeader = styled.div`
  & > * {
    margin: 0;
  }
`;
```

Reply:
Here is how my storybook:
```tsx
const CustomHeadingSubHeadingTemplate: Story<CardProps> = args => {
  const heading = (
    <Typography variant="subtitle1" color={Red.B74}>
      Custom Heading
    </Typography>
  );

  const subheading = (
    <Typography variant="subtitle2" color={Red.B65}>
      Custom Sub Heading
    </Typography>
  );

  return (
    <Card {...args} heading={heading} subheading={subheading}>
      <Card.Section>This is a section</Card.Section>
    </Card>
  );
};
```

Reply:
Found the root cause:
<img width="397" alt="image" src="https://user-images.githubusercontent.com/52077891/221115870-30d8afd6-7eec-4bd7-8880-43bd1bca6287.png">

Reply:
Do you think it is something that the user (who consume this component) need to handle themselves?

Reply:
If yes, I am good with removing the conditional (PRIMITIVE_TYPES)

Reply:
Looks like it's coming from the node passed to the `Card` which is `p`? I think since we can't predict what will be passed as heading, probably better for user to adjust the style while we can make sure the font style is what we want when wrapping them with Typography 🤔

Reply:
Ok if that so, I will delete this condition `PRIMITIVE_TYPES.includes(typeof subheading) ?`.

Reply:
An example to fix it on the user side,
```tsx
const CustomHeadingSubHeadingTemplate: Story<CardProps> = args => {
  const StyledTypography = styled(Typography)`
    margin: 0;
  `;

  const heading = (
    <StyledTypography variant="subtitle1" color={Red.B74}>
      Custom Heading
    </StyledTypography>
  );

  const subheading = (
    <StyledTypography variant="subtitle2" color={Red.B65}>
      Custom Sub Heading
    </StyledTypography>
  );

  return (
    <Card {...args} heading={heading} subheading={subheading}>
      <Card.Section>This is a section</Card.Section>
    </Card>
  );
};
```

Reply:
Removed